hpx/parallel/execution_policy.hpp

```diff
@@ -65,11 +111,302 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
+    static sequential_execution_policy const seq;
+
+    /// The class sequential_execution_policy is an execution policy type used
+    /// as a unique type to disambiguate parallel algorithm overloading and
+    /// require that a parallel algorithm's execution may not be parallelized.
+    template <typename Executor>
+    struct sequential_execution_policy_shim : sequential_execution_policy
+    {
+        /// The type of the executor associated with this execution policy
+        typedef Executor executor_type;
+
+        /// The category of the execution agents created by this execution
+        /// policy.
+        typedef typename executor_traits<executor_type>::execution_category
+            execution_category;
+
+        /// Rebind the type of executor used by this execution policy. The
+        /// execution category of Executor shall not be weaker than that of
+        /// this execution policy
+        template <typename Executor_>
+        struct rebind
+        {
+            /// The type of the rebound execution policy
+            typedef sequential_execution_policy_shim<Executor_> type;
+        };
+
+        /// Create a new sequential_task_execution_policy.
+        ///
+        /// \param tag          [in] Specify that the corresponding asynchronous
+        ///                     execution policy should be used
+        ///
+        /// \returns The new sequential_task_execution_policy_shim
+        ///
+        sequential_task_execution_policy_shim<Executor> operator()(
+            task_execution_policy_tag tag) const
+        {
+            return sequential_task_execution_policy_shim<Executor>(exec_);
+        }
+
+        /// Create a new sequential_execution_policy_shim from the given
+        /// executor
+        ///
+        /// \tparam Executor    The type of the executor to associate with this
+        ///                     execution policy.
+        ///
+        /// \param exec         [in] The executor to use for the
+        ///                     execution of the parallel algorithm the
+        ///                     returned execution policy is used with.
+        ///
+        /// \note Requires: is_executor<Executor>::value is true
+        ///
+        /// \returns The new parallel_task_execution_policy
+        ///
+        template <typename Executor_>
+        typename rebind_executor<sequential_execution_policy, Executor_>::type
+        on(Executor_& exec) const
+        {
+            BOOST_STATIC_ASSERT(is_executor<Executor_>::value);
+
+            typedef typename rebind_executor<
+                sequential_execution_policy, Executor_
+            >::type rebound_type;
+            return rebound_type(exec);
+        }
+
+        /// Return the associated executor object.
+        Executor& executor() { return exec_; }
+        /// Return the associated executor object.
+        Executor const& executor() const { return exec_; }
+
+    private:
+        /// \cond NOINTERNAL
```

Comment:
Is this missing friends from `template <typename> sequential_execution_policy_shim` and `sequential_execution_policy`?